minimalTest.js

```diff
@@ -301,16 +299,36 @@ function render() {
         );
       }, "");
 
+    const prevScrollTop = view.scrollTop;
+    const isAtBottom = view.scrollTop === 0;
     view.innerHTML = messageHistoryHTML;
 
-    // Autoscroll to new message, when scrollbar is at the bottom of chatbox
-    const isScrolledToBottom =
-      view.scrollHeight - view.clientHeight <= view.scrollTop + 1;
-
-    if (isScrolledToBottom) {
-      view.scrollTop = view.scrollHeight - view.clientHeight;
+    if (isAtBottom) {
+      view.scrollTop = 0;
+    } else {
+      view.scrollTop = prevScrollTop;
     }
   }
+
+  console.log(appMode);
+  if (appMode === "game") {
+    // Always show canvas in game mode
+    document.getElementById("canvas").style.display = "block";
+    document.getElementById("toggle_chat")?.classList.add("bg-slate-600");
+    document.getElementById("toggle_chat")?.classList.add("hover:bg-slate-700");
+    document.getElementById("toggle_chat")?.classList.remove("bg-green-400");
+    document
+      .getElementById("toggle_chat")
+      ?.classList.remove("hover:bg-green-500");
+  } else {
+    document.getElementById("canvas").style.display = "none";
+    document.getElementById("toggle_chat")?.classList.remove("bg-slate-600");
+    document
+      .getElementById("toggle_chat")
+      ?.classList.remove("hover:bg-slate-700");
+    document.getElementById("toggle_chat")?.classList.add("bg-green-400");
+    document.getElementById("toggle_chat")?.classList.add("hover:bg-green-500");
```

Comment:
```suggestion
    document.getElementById("toggle_chat").classList.add("bg-slate-600");
    document.getElementById("toggle_chat").classList.add("hover:bg-slate-700");
    document.getElementById("toggle_chat").classList.remove("bg-green-400");
    document
      .getElementById("toggle_chat")
      .classList.remove("hover:bg-green-500");
  } else {
    document.getElementById("canvas").style.display = "none";
    document.getElementById("toggle_chat").classList.remove("bg-slate-600");
    document
      .getElementById("toggle_chat")
      .classList.remove("hover:bg-slate-700");
    document.getElementById("toggle_chat").classList.add("bg-green-400");
    document.getElementById("toggle_chat").classList.add("hover:bg-green-500");
```
we should probably get rid of any `?` assertions because they mask errors, notice the first line doesnt throw an error, so if we make a typo in the element id or smth we will never see it in console

![image](https://user-images.githubusercontent.com/12636708/222045052-b94e0062-e03d-499b-842e-569eb796f3b6.png)

I guess the "right" way to get vscode to stop complaining with the minimal amount of `@ts-ignore` will be smth like

```
/** @type {HTMLElement} */
// @ts-ignore
const toggleChat = document.getElementById("toggle_chat");
```

but for now we can just tolerate all the red squigglies in the editor